### PR TITLE
Support for attach network to ports of paired TOR switch

### PR DESCRIPTION
**Pull request by praveenramoorthy**
Support for attach network to ports of paired TOR switch

### CODE REVIEW

**Comment by mikewiebe** [comment]
@praveenramoorthy Looks like some CI checks are still failing

**Comment by dsx1123** [comment]
LGTM

File: tests/integration/targets/dcnm_network/tests/dcnm/self-contained-tests/tor_ports_networks.yaml [hunk 1]
```diff
@@ -0,0 +1,366 @@
...
+- block:
+  - set_fact:
+      rest_path: "/rest/control/fabrics/{{ test_fabric }}"
+    when: controller_version == "11"
+  
+  - set_fact:
+      rest_path: "/appcenter/cisco/ndfc/api/v1/lan-fabric/rest/control/fabrics/{{ test_fabric }}"
+    when: controller_version >= "12"
+  
+  - name: SCALE - Verify if fabric is deployed.
```
Comment (mikewiebe) [inline tests/integration/targets/dcnm_network/tests/dcnm/self-contained-tests/tor_ports_networks.yaml:14]: *nit* Remove `SCALE`?

File: tests/integration/targets/dcnm_network/tests/dcnm/self-contained-tests/tor_ports_networks.yaml [hunk 2]
```diff
@@ -0,0 +1,366 @@
...
+    cisco.dcnm.dcnm_rest:
+      method: GET
+      path: "{{ rest_path }}"
+    register: result
+  
+  - assert:
+      that:
+      - 'result.response.DATA != None'
+  
+  - name: SCALE - Clean up any existing networks
```
Comment (mikewiebe) [inline tests/integration/targets/dcnm_network/tests/dcnm/self-contained-tests/tor_ports_networks.yaml:24]: *nit* Remove `SCALE`?

File: plugins/modules/dcnm_network.py [hunk 3]
```diff
@@ -551,6 +573,74 @@ def diff_for_attach_deploy(self, want_a, have_a, replace=False):
 
                         if want.get("isAttached") is not None:
                             if bool(have["isAttached"]) and bool(want["isAttached"]):
+                                torports_configured = False
+
+                                # Handle tor ports first if configured.
+                                if want.get("torports"):
```
Comment (mikewiebe) [inline plugins/modules/dcnm_network.py:579]: Quite a bit of nested logic here.  This is not something we should necessarily change now but just bringing it up for discussion.  Let's explore if we could improve the readability / long term maintenance by breaking this up into smaller helper functions.  What do you think?  Again, not a blocker for this PR.

File: tests/integration/targets/dcnm_network/tests/dcnm/self-contained-tests/tor_ports_networks.yaml (hunk 1 above)
Comment (praveenramoorthy) [reply tests/integration/targets/dcnm_network/tests/dcnm/self-contained-tests/tor_ports_networks.yaml:14]: Changed to TOR

File: tests/integration/targets/dcnm_network/tests/dcnm/self-contained-tests/tor_ports_networks.yaml (hunk 2 above)
Comment (praveenramoorthy) [reply tests/integration/targets/dcnm_network/tests/dcnm/self-contained-tests/tor_ports_networks.yaml:24]: Changed to TOR

File: plugins/modules/dcnm_network.py (hunk 3 above)
Comment (praveenramoorthy) [reply plugins/modules/dcnm_network.py:579]: Sure, we can do that for future commits

**Review by mikewiebe — APPROVED** [review]

**Pull request merged**